src/client/dfuse/il/int_posix.c

```diff
@@ -256,21 +365,15 @@ check_ioctl_on_open(int fd, struct fd_entry *entry, int flags, int status)
 
 	if (ioil_ioc.ioc_open_fd_count == 0) {
 
-		svcl = daos_rank_list_parse("0", ":");
-
-		rc = daos_pool_connect(il_reply.fir_pool, NULL, svcl,
-				       DAOS_PC_RW, &ioil_ioc.ioc_poh, NULL,
-				       NULL);
-		if (rc)
-			D_GOTO(drop_lock, 0);
+		rc = fetch_daos_handles(fd);
+		if (rc) {
+			DFUSE_LOG_INFO("fetch handles failed");
```

Comment:
why would this fail here?

Reply:
The major reason would be if the interception library was running as a different user or group to dfuse, in which case the ioctl would return with EPERM.  Not an issue today, but it'll become one when we support multiple users.

There are several other other issues as well, there's memory allocation (both here and in dfuse) and I suspect it's possible for the global handles to change size between the different ioctl calls which would also cause a failure, we could possibly attempt to re-try in that case.